tools/mpremote: Add metadata file and new subcommands for mip

aivarannamaa opened the pull request:
### Summary

1. Make `mpremote mip install` create or update a metadata file containing installation details
2. Use this impormation to implement `mpremote mip list` and `mpremote mip uninstall`.

### Testing

I tested on a Raspberry Pi Pico. I'll perform more thorough testing if the maintainers express their interest in this idea.

### Trade-offs and Alternatives

The discussion has been started at https://github.com/orgs/micropython/discussions/17534

Josverl commented:
I tend to check [other runs of the same GHA ,  `ports unix`](https://github.com/micropython/micropython/actions/workflows/ports_unix.yml) , and if there are others failing, that share the same base , then its time for one more rebase.

aivarannamaa committed:
tools/mpremote: Add metadata file and new subcommands for mip.

Make mip install create or update mip-packages.json file containing
installed packages metadata. Implement `list` and `uninstall` subcommands.

Signed-off-by: Aivar Annamaa <aivar.annamaa@gmail.com>
